### ISSUE TITLE
PostProcessing - unRAR, then convert (+ Solution)

### ISSUE THREAD

**Issue by traxinv** [mdhiggins/sickbeard_mp4_automator#401]
@mdhiggins So I had a bit of a problem with the uTorrent post processing as it does not automatically unrar the files it downloads. So instead of complaining I came up with a solution. Additionally, I only wanted it to process files when they are finished. Since I use a finite seeding rule this is pretty simple to predict when the post processing will occur. 

I have attached the program I wrote to unrar the files and then call the MP4 automator. Note i have not sorted out the sample delete issue. But i think we should maybe alter mp4_automator to ignore samples. 

Here is the command line as called from uTorrent:

RecUnRar.exe %L %T %D %K %F %I %N False False %S

L|Lable      - Torrent lable
T|Tracker    - Torrent tracker used to download
D|Directory  - Directory where torrent was downloaded
K|KindTor    - Kind of torrent (Single file vs MultiFile)
F|FileName   - File Name (if single file)
I|HEX        - Torrent hex encoded info hash
N|Title      - Title of Torrent

++Optional
DD|DeleteRAR - Delete the RAR files after decompress completes. (Default = True)
DS|DeleteSample - Delete any sample files if found. (Default = True) (This sometimes confuses the postprocessor)
S|State - State of the torrent it will only process the download when state is: 11 (Finished) - I am planning some more features here. 
Please

There are a few settings you need to configure in: RecUnRAR.exe.config

<add key="DirWinRar" value="C:\Program Files\WinRAR" />
<add key="PostProcess" value="C:\MediaServer\Software\Converter\uTorrentPostProcess.py" />
<add key="AppUnRar" value="UnRAR.exe" />
<add key="Dir7Zip" value="C:\MediaServer\Software\7-Zip" />
<add key="App7Zip" value="7z.exe" />

[RecUnRar.zip](https://github.com/mdhiggins/sickbeard_mp4_automator/files/90427/RecUnRar.zip)

**Comment by traxinv** [comment]
So ran into some problems with file paths that have spaces in... stupid me forgot about them. Added is an updated version. The command line for uTorrent has changed to:

RecUnRAR.exe "%L" "%T" "%D" "%K" "%F" "%I" "%N" FALSE FALSE "%S" 
[RecUnRAR v2.zip](https://github.com/mdhiggins/sickbeard_mp4_automator/files/97046/RecUnRAR.v2.zip)

**Comment by mdhiggins** [comment]
Probably won't ever formally implement this solution since it's not cross platform. It's low priority on the to do list since most of the media managers will step in and extract in the event of a compressed file. 

Also quite rare to see a compressed torrent

Either way the best option would be a python library that handles extraction if such a thing exists. 

Feel free to make a fork of this solution though I imagine some people may find this useful!

**Comment by traxinv** [comment]
So most of the movies etc are in compressed RAR files so I want to unrar the files before I pass it to the media managers as it will rename and move the file before starting the conversion which is a problem on my side. 

However, I am busy writing a python version. The C# version is just allot quicker for me since I work with it every day.
